File: src/estacionamento_java/Vaga.java
package estacionamento_java;

import java.time.LocalDateTime;

public class Vaga {

    private boolean ocupado;
    private String numero;
    private LocalDateTime incio;
    private Veiculo veiculo;

    public Vaga(String numero) {
        this.numero = numero;
        this.ocupado = false;
    }

    public void alocarVeiculo(Veiculo veiculo) {
        this.veiculo = veiculo;
        this.ocupado = true;
        this.incio = LocalDateTime.now();
    }
}

File: src/main/kotlin/Vaga.kt
import java.time.LocalDateTime

data class Vaga(
    var ocupado: Boolean = false,
    val numero: String,
    var incio: LocalDateTime? = null,
    var veiculo: Veiculo? = null
) {
    fun alocarVeiculo(veiculo: Veiculo?) {
        this.ocupado = true
        this.incio = LocalDateTime.now()
        this.veiculo = veiculo
    }
}
